kgtk/gt/embedding_utils.py

```diff
@@ -652,36 +681,78 @@ def dump_vectors(self, file_name, type_=None):
                     _ = f.write("\n")
 
     def print_vector(self, vectors, output_properties: str = "text_embedding",
-                     output_format="kgtk_format", save_embedding_sentence=False):
+                     output_format="kgtk_format", save_embedding_sentence=False, output_file=""):
         self._logger.debug("START printing the vectors")
         if output_format == "kgtk_format":
-            # TODO: This should be comverted to use KgtkWriter
-            print("node\tproperty\tvalue\n", end="")
+            if output_file != "-":
+                output_mode = KgtkWriter.Mode.NONE
+                ew: KgtkWriter = KgtkWriter.open(self.column_names,
+                                            output_file,
+                                            require_all_columns=False,
+                                            prohibit_extra_columns=True,
+                                            fill_missing_columns=True,
+                                            mode=output_mode,
+                                            output_format='kgtk',
+                                            output_column_names=self.column_names,
+                                            no_header=False,
+                                            error_file=self.error_file,
+                                            verbose=True,
+                                            very_verbose=False)
             all_nodes = list(self.vectors_map.keys())
             ten_percent_len = math.ceil(len(vectors) / 10)
             for i, each_vector in enumerate(vectors):
                 if i % ten_percent_len == 0:
                     percent = i / ten_percent_len * 10
                     self._logger.debug("Finished {}%".format(percent))
-                print("{}\t{}\t".format(all_nodes[i], output_properties), end="")
+                node_id = all_nodes[i]
+                props = output_properties
+                embedding = ""
                 for each_dimension in each_vector[:-1]:
-                    print(str(each_dimension) + ",", end="")
-                print(str(each_vector[-1]))
+                    embedding += str(each_dimension) + ","
+                embedding += str(each_vector[-1])
+                if output_file == "-":
+                    print("{}\t{}\t{}".format(node_id, props, embedding))
+                else:
+                    ew.write_kgtk([node_id, props, embedding])
                 if save_embedding_sentence:
-                    print("{}\t{}\t{}".format(all_nodes[i], "embedding_sentence",
+                    if output_file == "-":
+                        print("{}\t{}\t{}\n".format(all_nodes[i], "embedding_sentence",
```

Comment:
Check this out https://github.com/usc-isi-i2/kgtk/blob/master/kgtk/io/kgtkwriter.py#L165

You dont have to handle "-" case yourself. KGTKWriter handles it

Reply:
Aah alright